codecs/avif/enc/avif_enc.cpp

```diff
@@ -69,14 +73,34 @@ val encode(std::string buffer, int width, int height, AvifOptions options) {
   avifImageRGBToYUV(image, &srcRGB);
 
   avifEncoder* encoder = avifEncoderCreate();
+
```

Comment:
Change line 67 from:

`  uint8_t* rgba = (uint8_t*)buffer.c_str();`

to:

`  const uint8_t* rgba = (const uint8_t*)buffer.data();`

or better yet:

`  const uint8_t* rgba = reinterpret_cast<const uint8_t*>(buffer.data());`

Reply:
Why's it better? (I'm a C/C++ novice)

Reply:
I just realized that we do need a non-const pointer here because of a limitation of libavif's C API. So I changed my suggestion to:

`uint8_t* rgba = (uint8_t*)buffer.data();`

or

`uint8_t* rgba = reinterpret_cast<uint8_t*>(const_cast<char*>(buffer.data()));`

Detail:

The `(type)` cast is called a C-style cast in C++. C++ supports C-style casts for compatibility with C, but C++ casts, such as `static_cast<type>()`, `const_cast<type>()`, and `reinterpret_cast<type>()`, are preferred. Each kind of C++ cast is intended for a particular purpose, so it is clear why we are casting.

For example, buffer.c_str() returns `const char*`:

https://en.cppreference.com/w/cpp/string/basic_string/c_str

The `(uint8_t*)` cast not only changes the pointer type but also removes the `const`. C++ casts force you to use two casts to perform these two operations: `const_cast<char*>()` to remove the `const`, followed by `reinterpret_cast<uint8_t*>` to change the pointer type.

In this case, because of the verbosity of C++ casts, you may consider it a plus to use the C-style cast. But in general C++ casts are preferred.

As for `.c_str()` vs. `.data()`, the `.data()` method is not required to return a buffer with a terminating `NUL` character, so it could be more efficient than the `.c_str()` method, and is semantically more correct in this case (because `buffer` contains binary data, not a text string). In practice `.data()` and `.c_str()` often have the same implementation.